torchmetrics/functional/text/eed.py

```diff
@@ -0,0 +1,446 @@
+
+    Args:
+        scores: sum of individual sentence scores as a tensor
+        total_num_sentences: number of sentences as a tensor
+
+    Returns:
+        average of scores as a tensor
+    """
+    if scores == tensor(0.0):
+        return scores
+
+    average = scores / total_num_sentences
+    return average
+
+
+def _preprocess_sentences(
+    reference_corpus: Sequence[Union[str, Sequence[str]]],
+    hypothesis_corpus: Union[str, Sequence[str]],
+    language: Union[Literal["en"], Literal["ja"]],
+) -> Tuple[List[str], List[str]]:
+    """Proprocess strings according to language requirements.
+
+    Args:
+        reference_corpus: An iterable of iterables of reference corpus.
+        hypothesis_corpus: An iterable of hypothesis corpus.
+        language: Language used in sentences. Only supports English (en) and Japanese (ja) for now. Defaults to en
+
+    Returns:
+        Tuple of lists that contain the cleaned strings for reference_corpus and hypothesis_corpus
+
+    Raises:
+        ValueError: If a different language than 'en" or 'ja' is used
+        ValueError: If length of reference_corpus not equal to length of hypothesis_corpus
+        ValueError: If objects in reference and hypothesis corpus are not strings
+    """
+    # sanity checks
+    reference_corpus, hypothesis_corpus = _validate_inputs(reference_corpus, hypothesis_corpus)
+
+    # preprocess string
+    if language == "en":
+        preprocess_function = _preprocess_en
+    elif language == "ja":
+        preprocess_function = _preprocess_ja
+    else:
+        raise ValueError(f"Expected argument `language` to either be `en` or `ja` but got {language}")
+
+    reference_corpus = [[preprocess_function(ref) for ref in reference] for reference in reference_corpus]
+    hypothesis_corpus = [preprocess_function(hyp) for hyp in hypothesis_corpus]
+
+    return reference_corpus, hypothesis_corpus
+
+
+def _compute_sentence_statistics(
+    reference_words: List[List[str]],
+    hypothesis: List[str],
+    alpha: float = 2.0,
+    rho: float = 0.3,
+    deletion: float = 0.2,
+    insertion: float = 1.0,
+) -> Tuple[Tensor, Tensor]:
+    """Compute scores for EED.
+
+    Args:
+        reference_words:
+            An iterable of reference words
+        hypothesis:
+            A sentence
+        alpha:
+            optimal jump penalty, penalty for jumps between characters
+        rho:
+            coverage cost, penalty for repetition of characters
+        deletion:
+            penalty for deletion of character
+        insertion:
+            penalty for insertion or substitution of character
+
+    Returns:
+        Tuple of scores and number of sentences as floats
+    """
+    scores = 0.0
+    num_sentences = 0.0
+
+    for reference in reference_words:
+        score = _eed_function(reference, hypothesis, alpha, rho, deletion, insertion)
+        scores += score
+        num_sentences += 1.0
+
+    return scores, num_sentences
```

Comment:
I am not sure if this is correct. You should take the average sentence-level score across multiple references already here and not consider all scores for the final calculation.
Moreover, I would also allow a user to take either an average or minimum EED per hypothesis when multiple references are provided as there is no direct in the original paper how to handle the cases with multiple references :]

Reply:
Ah, so that's how it's handled. Yeah, the paper didn't have anything on handling multiple references.